modules/tensor_mechanics/include/actions/TensorMechanicsAction.h

```diff
@@ -23,17 +23,35 @@ class TensorMechanicsAction : public Action
 
 protected:
   virtual std::string getKernelType();
-  virtual InputParameters getParameters(std::string type);
+  virtual InputParameters getKernelParameters(std::string type);
 
+  ///@{ displacement variables
   std::vector<NonlinearVariableName> _displacements;
   unsigned int _ndisp;
-
   std::vector<VariableName> _coupled_displacements;
+  ///@}
 
+  ///@{ residual debugging
   std::vector<AuxVariableName> _save_in;
   std::vector<AuxVariableName> _diag_save_in;
+  ///@}
 
   Moose::CoordinateSystemType _coord_system;
+
+  /// if this vector is not empty the variables, kernels and materials are restricted to these subdomains
+  std::vector<SubdomainName> _subdomain_names;
+
+  /// strain formulation
+  enum class Strain { Small, Finite } _strain;
+
+  /// strain formulation
+  enum class StrainAndIncrement { SmallTotal, FiniteTotal, SmallIncremental, FiniteIncremental } _strain_and_increment;
+
+  /// use a plane strain formulation
+  enum class PlaneStrain { None, Weak, Generalized } _plane_strain;
```

Comment:
I don't think this is the right way to handle the plane strain options. The `Weak` option isn't plane strain at all (it's plane stress), and there's not option in this enum for regular plane strain. It doesn't look like things are really set up to do anything if the user specifies this anyway, so I'd say we should just remove this option and anything that uses it for the time being.

It seems like we want a single parameter that we can use to specify how the out of plane behavior is handled for 2d (and 1d?) models. The options would be plane stress, plane strain, axisymmetric, and generalized plane strain. I don't think `plane_strain` is the right name for that parameter.